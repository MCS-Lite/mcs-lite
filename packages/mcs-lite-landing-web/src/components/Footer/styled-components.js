import styled from 'styled-components';
import { Column, withBreakpoints } from 'hedron';

export const Container = styled.footer`
  background-color: ${props => props.theme.color.black};
`;

export const StyledColumn = styled(Column)`
  padding-top: 10px;
  padding-bottom: 10px;
`;

export const IconWrapper = styled.div`
  display: flex;

  a {
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  a + a {
    margin-left: 20px;
  }

  img {
    height: 32px;
    width: 32px;
  }
`;

export const RWDWrapper = withBreakpoints(styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;

  @media (max-width: ${props => props.breakpoints.sm}px) {
    flex-direction: column-reverse;
    text-align: center;

    ${IconWrapper} {
      margin-bottom: 5px;
    }
  }
`);
