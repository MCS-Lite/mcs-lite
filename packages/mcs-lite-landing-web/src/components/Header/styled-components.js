import styled from 'styled-components';
import { Link } from 'react-router';
import { Column, withBreakpoints } from 'hedron';
import Nav from 'mcs-lite-ui/lib/LandingHeader/Nav';
import NavItem from 'mcs-lite-ui/lib/LandingHeader/NavItem';

export const StyledColumn = styled(Column)`
  padding-top: 0;
  padding-bottom: 0;
  display: flex;
  justify-content: space-between;
`;

export const HiddenForPreRenderTrick = styled.div`
  visibility: hidden;
  display: none;
`;

export const StyledLink = styled(Link)`
  text-decoration: none;
`;

export const LogoImage = styled.img`
  height: 28px;
  width: auto;
`;

export const DesktopNav = withBreakpoints(styled(Nav)`
  ${'' /* For SEO */}
  display: none;

  @media (min-width: ${props => props.breakpoints.sm}px) {
    display: inherit;
  }
`);

export const MobileNav = styled(Nav)`
  ${NavItem} {
    padding-right: 0;
  }
`;
