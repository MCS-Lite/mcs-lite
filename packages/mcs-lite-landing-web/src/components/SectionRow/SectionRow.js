import styled from 'styled-components';
import { Row } from 'hedron';

const SectionRow = styled(Row)`
  padding: 80px 0;
`;

export default SectionRow;
