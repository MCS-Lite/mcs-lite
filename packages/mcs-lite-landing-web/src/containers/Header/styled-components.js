import styled from 'styled-components';
import { Column } from 'hedron';
import P from 'mcs-lite-ui/lib/P';
import A from 'mcs-lite-ui/lib/A';
import { MenuItem, Menu } from 'mcs-lite-ui/lib/Menu';
import IconFold from 'mcs-lite-icon/lib/IconFold';

export const HEIGHT = 50;
const HEADER_ZINDEX = 1;

export const Container = styled.header`
  height: ${HEIGHT}px;
  background: ${props => props.theme.color.grayLight};
`;

// prettier-ignore
export const Fixed = styled.div`
  z-index: ${HEADER_ZINDEX};
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  background: ${props => props.theme.color.grayLight};
  box-shadow: ${props => (props.isTop ? 'none' : '0 1px 0 0 rgba(0, 0, 0, 0.05)')};
  transition: box-shadow cubic-bezier(0.47, 0, 0.75, 0.72) 0.2s;
`;

export const StyledColumn = styled(Column)`
  padding-top: 0;
  padding-bottom: 0;
  display: flex;
  justify-content: space-between;
`;

export const StyledA = styled(A)`
  height: ${HEIGHT}px;
  display: flex;
  align-items: center;
  cursor: pointer;
  text-decoration: none;
  color: ${props => props.theme.color.black};
`;

export const StyledP = styled(P)`
  margin-right: 5px;
`;

export const Right = styled.div`
  display: flex;

  > ${StyledA} {
    padding: 0 15px;
  }

  > ${StyledA}:last-child {
    padding-right: 0;
  }
`;

export const StyledMenu = styled(Menu)`
  ${MenuItem} {
    width: 100%;
  }

  a {
    text-decoration: none;
  }
`;

export const HiddenForPreRenderTrick = styled.div`
  visibility: hidden;
  display: none;
`;

export const StyledIconFold = styled(IconFold)`
  transform: rotate(${props => (props.isShow ? -180 : 0)}deg);
  transition: transform 0.4s cubic-bezier(0.68, -0.55, 0.27, 1.55);
`;
