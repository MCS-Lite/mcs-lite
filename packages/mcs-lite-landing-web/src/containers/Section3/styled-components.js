import React from "react";
import PropTypes from "prop-types";
import { Row } from "hedron";
import styled from "styled-components";
import ScrollOverPack from "rc-scroll-anim/lib/ScrollOverPack";
import TweenOne from "rc-tween-one";
import LazyloadOnce from "mcs-lite-ui/lib/LazyloadOnce";

const HEIGHT = 80;

export const ImageWrapper = styled.div`
  width: ${HEIGHT}px;
  height: ${HEIGHT}px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  border-radius: 3px;
`;

export const LazyloadIcon = ({ children }) => (
  <LazyloadOnce height={HEIGHT}>
    <ScrollOverPack playScale={0.2}>
      <TweenOne
        key="ImageWrapper"
        animation={{ opacity: 1, y: 10 }}
        style={{ opacity: 0, transform: "translateY(-30px)" }}
      >
        <ImageWrapper>{children}</ImageWrapper>
      </TweenOne>
    </ScrollOverPack>
  </LazyloadOnce>
);

LazyloadIcon.displayName = "LazyloadIcon";
LazyloadIcon.propTypes = {
  children: PropTypes.node.isRequired
};

export const StyledRow = styled(Row)`
  > * {
    &:nth-of-type(4n + 1) ${ImageWrapper} {
      background-color: #28b2e6;
    }

    &:nth-of-type(4n + 2) ${ImageWrapper} {
      background-color: #feb439;
    }

    &:nth-of-type(4n + 3) ${ImageWrapper} {
      background-color: #36c1ca;
    }

    &:nth-of-type(4n) ${ImageWrapper} {
      background-color: #ff4e84;
    }
  }
`;
