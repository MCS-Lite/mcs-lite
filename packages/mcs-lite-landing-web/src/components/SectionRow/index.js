import { pure } from 'recompose';
import SectionRow from './SectionRow';

export default pure(SectionRow);
