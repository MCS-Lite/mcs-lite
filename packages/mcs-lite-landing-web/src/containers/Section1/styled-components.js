import styled from 'styled-components';
import { Column, withBreakpoints } from 'hedron';
import SectionRow from '../../components/SectionRow';

export const StyledSectionRow = styled(SectionRow)`
  background-image: linear-gradient(-180deg, #FAFAFA 0%, #F1F2F7 100%);
  padding-bottom: 0;
  overflow: hidden;
`;

export const RWDCenterWrapper = withBreakpoints(styled.div`

  @media (max-width: ${props => props.breakpoints.md}px) {
    text-align: center;
  }
`);

export const StyledImageColumn = styled(Column)`
  padding-bottom: 0;
`;

export const ScreenImageMobile = withBreakpoints(styled.img`
  display: none;
  height: auto;
  width: 100%;
  box-shadow: 0 5px 8px 0 rgba(0, 0, 0, 0.1);

  @media (max-width: ${props => props.breakpoints.sm}px) {
    display: block;
  }
`);

export const ImageLayerWrapper = withBreakpoints(styled.div`
  position: relative;
  height: ${props => props.height}px;
  display: flex;

  > * {
    position: absolute;
  }

  @media (max-width: ${props => props.breakpoints.md}px) {
    align-items: center;
    justify-content: center;
  }
`);

export const BackgroundImage = styled.img`
  height: 100%;
  width: auto;
  box-shadow: 0 5px 8px 0 rgba(0, 0, 0, 0.1);
`;
