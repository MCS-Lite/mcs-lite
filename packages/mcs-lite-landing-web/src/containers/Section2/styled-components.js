import styled from 'styled-components';
import SectionRow from '../../components/SectionRow';

export const StyledSectionRow = styled(SectionRow)`
  background-image: linear-gradient(-180deg, #FFFFFF 0%, #FDFDFD 47%, #FAFAFA 100%);
`;

export const MacImage = styled.img`
  height: 100%;
  width: auto;
  max-width: 100%;
`;
