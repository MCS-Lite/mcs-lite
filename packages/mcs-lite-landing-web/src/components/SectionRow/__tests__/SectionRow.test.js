import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ThemeProvider } from 'styled-components';
import landingTheme from '../../../utils/landingTheme';
import SectionRow from '../SectionRow';

it('should renders <SectionRow> correctly', () => {
  const wrapper = mount(
    <ThemeProvider theme={landingTheme}>
      <SectionRow>
        children
      </SectionRow>
    </ThemeProvider>,
  );

  const tree = toJson(wrapper.find(SectionRow));

  expect(tree).toMatchSnapshot();
});
