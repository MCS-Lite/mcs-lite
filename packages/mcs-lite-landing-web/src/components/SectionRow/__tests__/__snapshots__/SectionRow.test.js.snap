// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders <SectionRow> correctly 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 {
  margin: 0 auto;
  max-width: 100%;
  width: 960px;
}

.c0 {
  padding: 60px 0;
  background-color: #FFFFFF;
}

<SectionRow>
  <styled.div>
    <div
      className="c0"
    >
      <Styled(PageContainer)
        width="960px"
      >
        <PageContainer
          className="c1"
          width="960px"
        >
          <div
            className="c1"
            width="960px"
          >
            <Styled(RowContainer)>
              <RowContainer
                className="c2"
                divisions={12}
              >
                <section
                  className="c2"
                >
                  children
                </section>
              </RowContainer>
            </Styled(RowContainer)>
          </div>
        </PageContainer>
      </Styled(PageContainer)>
    </div>
  </styled.div>
</SectionRow>
`;
